File: latmo2php/Human.php
<?php

/*Membuat kelas Human*/
class Human{
    /* class Human memiliki 3 atribut yaitu NIK, nama, dan jenis kelamin */
    private $NIK;
    private $nama;
    private $jenisKelamin;

    /* Konstruktor yang parameter nya berisi 3 atribut private */
    function __construct($nik, $nama, $jenisKelamin){
        $this->NIK = $nik;
        $this->nama = $nama;
        $this->jenisKelamin = $jenisKelamin;
    }

    // Get NIK
    public function getNIK(){
        return $this->NIK;
    }
    // Get Nama
    public function getNama(){
        return $this->nama;
    }
    // Get Jenis Kelamin
    public function getJenisKelamin(){
        return $this->jenisKelamin;
    }

    // jangan lupa destruktor ditiap class
    function __destruct()
    {
        
    }
}

/*  class Sivitas Akademik anak dari class Human*/
class SivitasAkademik extends Human{
     /* class Sivitas Akademik memiliki 2 atribut yaitu asal unic dan email edu, class ini merupakan anak dari class Human */
    private $asalUniv;
    private $emailEdu;

    // Konstruktor yang parameternya berisi atribut orantuanya sekaligus atribut utamanya, agar class sivitasakademik dapat menggunakan atribut yang ada di orangtuanya dan juga merupakan syarat agar class sivitas akademik menjadi anaknya dari Human
    function __construct($nik, $nama, $jenisKelamin, $asalUniv, $emailEdu)
    {
        parent::__construct($nik, $nama, $jenisKelamin); // disini di panggil lagi buat jalanin fungsi gitu pokonya biar atributnya bisa di akses
        $this->asalUniv = $asalUniv;
        $this->emailEdu = $emailEdu;
    }

    // Get Asal Univ
    public function getAsaluniv()
    {
        return $this->asalUniv;
    }
    // Get Email Edu
    public function getEmail()
    {
        return $this->emailEdu;
    }
    // Destructor class sivitas akademik
    function __destruct()
    {
        
    }
}

/* class Mahasiswa anak dari class SivitasAkademik */
class Mahasiswa extends SivitasAkademik{
    // class ini memiliki 4 tribut utama yaitu NIM, fakultas, prodi, dan foto
    private $NIM;
    private $fakultas;
    private $prodi;
    private $foto;

    // Konstruktor yang parameternya merupakan semua atribut darii orangtua dan kakeknya si class mahasiswa karena memang aturannya seperti itu kalo mau jadi anak
    function __construct($nik, $nama, $jenisKelamin, $asalUniv, $emailEdu, $Nim, $fakultas, $prodi, $foto)
    {
        parent::__construct($nik, $nama, $jenisKelamin, $asalUniv, $emailEdu); // disini juga sama buat ngejalanin fungsi class gitu biar atribut nya bisa diakses
        $this->NIM = $Nim;
        $this->fakultas = $fakultas;
        $this->prodi = $prodi;
    }

    // Get Nim
    public function getNIM()
    {
        return $this->NIM;
    }
    // Get fakultas
    public function getFakultas()
    {
        return $this->fakultas;
    }
    // Get Prodi
    public function getProdi()
    {
        return $this->prodi;
    }
    // Get path Foto
    public function getFoto(){
        return $this->foto;
    }
    // Konstruktor class mahasiswa
    function __destruct()
    {
        
    }
}
?>

File: latmo2py/Human.py
class Human:
    def __init__(self, NIK, nama, jenis_kelamin):
        self.NIK = NIK
        self.nama = nama
        self.jenis_kelamin = jenis_kelamin
        
    # get NIK
    def get_NIK(self):
        return self.__NIK
    # set NIK
    def set_NIK(self, NIK):
        self.__NIK = NIK
    # get nama
    def get_nama(self):
        return self.__nama
    # set nama
    def set_nama(self, nama):
        self.__nama = nama
    # get jenis_kelamin
    def get_jenis_kelamin(self):
        return self.__jenis_kelamin
    # set jenis_kelamin
    def set_jenis_kelamin(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin

class SivitasAkademik(Human):
    def __init__(self, NIK, nama, jenis_kelamin, asal_universitas, email_edu):
        super().__init__(NIK, nama, jenis_kelamin)
        self.asal_universitas = asal_universitas
        self.email_edu = email_edu
    
    # get asal_univ   
    def get_asal_univ(self):
        return self.__asal_univ
    # set asal_univ
    def set_asal_univ(self, asal_univ):
        self.__asal_univ = asal_univ
    # get email_edu
    def get_email_edu(self):
        return self.__email_edu
    # set email_edu
    def set_email_edu(self, email_edu):
        self.__email_edu = email_edu

class Mahasiswa(SivitasAkademik):
    def __init__(self, NIK, nama, jenis_kelamin, asal_universitas, email_edu, NIM, fakultas, prodi):
        super().__init__(NIK, nama, jenis_kelamin, asal_universitas, email_edu)
        self.NIM = NIM
        self.fakultas = fakultas
        self.prodi = prodi
        
    # get NIM
    def get_NIM(self):
        return self.__NIM
    # set NIM
    def set_NIM(self, NIM):
        self.__NIM = NIM
    # get fakultas
    def get_fakultas(self):
        return self.__fakultas
    # set fakultas
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
    # get jenis_kelamin
    def get_prodi(self):
        return self.__prodi
    # set prodi
    def set_prodi(self, prodi):
        self.__prodi = prodi

    # ini ada method tampil but nampilin data. iseng nyobain eh bisa hehe
    def tampil(self):
        print("Data Mahasiswa")
        print("NIK: ", self.NIK)
        print("Nama: ", self.nama)
        print("Jenis Kelamin: ", self.jenis_kelamin)
        print("Asal Universitas: ", self.asal_universitas)
        print("Email Edu: ", self.email_edu)
        print("NIM: ", self.NIM)
        print("Fakultas: ", self.fakultas)
        print("Program Studi: ", self.prodi)

# contoh penggunaan

